node/src/service.rs

```diff
@@ -66,10 +67,13 @@ pub async fn new_partial(
 	ServiceError,
 > {
 	let mc_follower_metrics = register_metrics_warn_errors(config.prometheus_registry());
-	let data_sources = crate::main_chain_follower::create_cached_main_chain_follower_data_sources(
-		mc_follower_metrics.clone(),
-	)
-	.await?;
+	let data_sources = task::block_in_place(|| {
+		config.tokio_handle.block_on(
+			crate::main_chain_follower::create_cached_main_chain_follower_data_sources(
+				mc_follower_metrics.clone(),
+			),
+		)
+	})?;
```

Comment:
We have lost nice, quick termination by signal. It blocks for one minute in current settings. This happens when there is no postgres available.